lib/src/RealIpResolver.cc

```diff
@@ -0,0 +1,201 @@
+/**
+ *
+ *  @file RealIpResolver.cc
+ *  @author Nitromelon
+ *
+ *  Copyright 2022, Nitromelon. All rights reserved.
+ *  https://github.com/drogonframework/drogon
+ *  Use of this source code is governed by a MIT license
+ *  that can be found in the License file.
+ *
+ *  Drogon
+ *
+ */
+
+#include <drogon/drogon.h>
+#include <trantor/utils/Logger.h>
+#include <drogon/plugins/RealIpResolver.h>
+
+using namespace drogon;
+using namespace drogon::plugin;
+
+struct XForwardedForParser
+{
+    explicit XForwardedForParser(std::string value)
+        : value_(std::move(value)), start_(value_.c_str()), len_(value_.size())
+    {
+    }
+
+    std::string getNext()
+    {
+        if (len_ == 0)
+        {
+            return {};
+        }
+        // Skip trailing separators
+        const char* cur;
+        for (cur = start_ + len_ - 1; cur > start_; --cur, --len_)
+        {
+            if (*cur != ' ' && *cur != ',')
+            {
+                break;
+            }
+        }
+        for (; cur > start_; --cur)
+        {
+            if (*cur == ' ' || *cur == ',')
+            {
+                ++cur;
+                break;
+            }
+        }
+        std::string ip{cur, len_ - (cur - start_)};
+        len_ = cur == start_ ? 0 : cur - start_ - 1;
```

Comment:
I think this assignment is unneeded as the variable isn't used and it returns afterwards

Reply:
The `getNext()` methods will be called multiple times until `len_` reach 0

Reply:
Codes to parse x-forwarded-for are learned from nginx

https://github.com/nginx/nginx/blob/f7ea8c76b55f730daa3b63f5511feb564b44d901/src/http/ngx_http_core_module.c#L2778-L2821